pumpkin/src/command/commands/cmd_clear.rs

```diff
@@ -0,0 +1,129 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use pumpkin_core::text::color::NamedColor;
+use pumpkin_core::text::TextComponent;
+
+use crate::command::args::arg_entities::EntitiesArgumentConsumer;
+use crate::command::args::{Arg, ConsumedArgs};
+use crate::command::tree::CommandTree;
+use crate::command::tree_builder::{argument, require};
+use crate::command::{CommandExecutor, CommandSender, InvalidTreeError};
+use crate::entity::player::Player;
+use InvalidTreeError::InvalidConsumptionError;
+
+const NAMES: [&str; 1] = ["clear"];
+const DESCRIPTION: &str = "Clear yours or targets inventory.";
+
+const ARG_TARGET: &str = "target";
+
+async fn clear_player(target: &Arc<Player>) -> usize {
+    let mut inventory = target.inventory.lock().await;
+
+    // There may be a better way to do this with something like all_slots() or .clear on Player.inventory
+    let mut items_count: usize = 0;
+    for slot in 0..=45 {
```

Comment:
You should use inventory.all_slots_mut() and then deref setting to None

Reply:
set_container_content function ?